packages/scaffold/src/partials/w3m-swap-details/index.ts

```diff
@@ -1,40 +1,66 @@
 import { html, LitElement } from 'lit'
-import { property } from 'lit/decorators.js'
+import { property, state } from 'lit/decorators.js'
 import styles from './styles.js'
 import { UiHelperUtil, customElement } from '@web3modal/ui'
 import { NumberUtil } from '@web3modal/common'
-import { NetworkController } from '@web3modal/core'
+import { ConstantsUtil, NetworkController, SwapController } from '@web3modal/core'
+
+// -- Constants ----------------------------------------- //
+const slippageRate = ConstantsUtil.CONVERT_SLIPPAGE_TOLERANCE
 
 @customElement('w3m-swap-details')
 export class WuiSwapDetails extends LitElement {
   public static override styles = [styles]
 
+  private unsubscribe: ((() => void) | undefined)[] = []
+
   // -- State & Properties -------------------------------- //
-  @property() public networkName = NetworkController.state.caipNetwork?.name
+  @state() public networkName = NetworkController.state.caipNetwork?.name
 
   @property() public detailsOpen = false
 
-  @property() public sourceTokenSymbol?: string
+  @state() public sourceToken = SwapController.state.sourceToken
+
+  @state() public sourceTokenPrice = SwapController.state.sourceTokenPrice
+
+  @state() public toToken = SwapController.state.toToken
 
-  @property() public sourceTokenPrice?: number
+  @state() public toTokenAmount = SwapController.state.toTokenAmount
```

Comment:
Since this component is a partial, it could reach SwapController and we don't have to pass properties. This makes the using it much easier